server/events/vcs/gitea/client.go

```diff
@@ -0,0 +1,468 @@
+func (c *GiteaClient) PullIsApproved(repo models.Repo, pull models.PullRequest) (models.ApprovalStatus, error) {
+	page := 0
+	nextPage := 1
+
+	approvalStatus := models.ApprovalStatus{
+		IsApproved: false,
+	}
+
+	listOptions := gitea.ListPullReviewsOptions{
+		ListOptions: gitea.ListOptions{
+			Page:     1,
+			PageSize: 100,
+		},
+	}
+
+	for page < nextPage {
+		page = +1
+		listOptions.ListOptions.Page = page
+		pullReviews, resp, err := c.giteaClient.ListPullReviews(repo.Owner, repo.Name, int64(pull.Num), listOptions)
+
+		if err != nil {
+			return approvalStatus, err
+		}
+
+		for _, review := range pullReviews {
+			if review.State == gitea.ReviewStateApproved {
+				approvalStatus.IsApproved = true
+				approvalStatus.ApprovedBy = review.Reviewer.UserName
+				approvalStatus.Date = review.Submitted
+
+				return approvalStatus, nil
+			}
+		}
+
+		nextPage = resp.NextPage
+
+		// Emergency break after 500 pages
+		if page >= 500 {
+			break
+		}
+	}
+
+	return approvalStatus, nil
+}
+
+// PullIsMergeable returns true if the pull request is mergeable
+func (c *GiteaClient) PullIsMergeable(repo models.Repo, pull models.PullRequest, vcsstatusname string) (bool, error) {
+	pullRequest, _, err := c.giteaClient.GetPullRequest(repo.Owner, repo.Name, int64(pull.Num))
+
+	if err != nil {
+		return false, err
+	}
+
+	return pullRequest.Mergeable, nil
+}
+
+// UpdateStatus updates the commit status to state for pull. src is the
+// source of this status. This should be relatively static across runs,
+// ex. atlantis/plan or atlantis/apply.
+// description is a description of this particular status update and can
+// change across runs.
+// url is an optional link that users should click on for more information
+// about this status.
+func (c *GiteaClient) UpdateStatus(repo models.Repo, pull models.PullRequest, state models.CommitStatus, src string, description string, url string) error {
+	giteaState := gitea.StatusFailure
+
+	switch state {
+	case models.PendingCommitStatus:
+		giteaState = gitea.StatusPending
+	case models.SuccessCommitStatus:
+		giteaState = gitea.StatusSuccess
+	case models.FailedCommitStatus:
+		giteaState = gitea.StatusFailure
+	}
+
+	newStatusOption := gitea.CreateStatusOption{
+		State:       giteaState,
+		TargetURL:   url,
+		Description: description,
+	}
+
+	_, _, err := c.giteaClient.CreateStatus(repo.Owner, repo.Name, pull.HeadCommit, newStatusOption)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DiscardReviews discards / dismisses all pull request reviews
+func (c *GiteaClient) DiscardReviews(repo models.Repo, pull models.PullRequest) error {
+	page := 0
+	nextPage := 1
+
+	dismissOptions := gitea.DismissPullReviewOptions{
+		Message: "Dismissed by Atlantis",
+	}
+
+	listOptions := gitea.ListPullReviewsOptions{
+		ListOptions: gitea.ListOptions{
+			Page:     1,
+			PageSize: 100,
```

Comment:
hard coded, please see previous comments

Reply:
Added --gitea-page-size, defaulting to 30 based on Gitea docs.